h/views/admin_users.py

```diff
@@ -137,21 +130,34 @@ def delete_user(request, user):
     """
     Deletes a user with all their group memberships and annotations.
 
-    Raises UserDeletionError when deletion fails with the appropriate error
-    message.
+    Any groups created by the user, which are empty after deleting the user's
+    own annotations, are removed. Other groups created by the user are kept.
     """
 
-    if models.Group.created_by(request.db, user).count() > 0:
-        raise UserDeletionError('Cannot delete user who is a group creator.')
-
-    user.groups = []
+    # Delete the user's annotations.
     annotations = request.db.query(models.Annotation) \
                             .filter_by(userid=user.userid)
     for annotation in annotations:
         storage.delete_annotation(request.db, annotation.id)
         event = AnnotationEvent(request, annotation.id, 'delete')
         request.notify_after_commit(event)
 
+    # Delete groups created by this user which are now empty.
+    empty_groups = []
+    for group in user.groups:
+        anns = request.db.query(models.Annotation) \
+                         .filter_by(groupid=group.pubid,
+                                    deleted=False)
+        if anns.count() > 0:
+            group.creator = None
```

Comment:
Currently:

* The public group is I think the only group that has creator unset

* If the creator of a group leaves that group their membership entry is removed from the `user_group` table, so they're no longer a member, but their user id remains as `creator_id` in the group table

* `group.creator_id` is a foreign key to `user` in the DB, so we can't delete a user account and leave their `id` in `group.creator_id`.

So I guess that, if we don't want to actually delete the group, then having a group whose `creator_id` is `NULL` is the only option. But this introduces a new case - groups with `NULL` `creator_id` because the creator's user account was deleted - when previously the public group was the only group without a creator.